addons/l10n_it_edi/models/ir_attachment.py

```diff
@@ -1,49 +1,93 @@
 # -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
 from odoo import api, models
 from odoo.addons.l10n_it_edi.tools.remove_signature import remove_signature
 
 from lxml import etree
 import logging
+import re
 
 _logger = logging.getLogger(__name__)
 
+FATTURAPA_FILENAME_RE = "[A-Z]{2}[A-Za-z0-9]{2,28}_[A-Za-z0-9]{0,5}.((?i:xml.p7m|xml))"
+
 
 class IrAttachment(models.Model):
     _inherit = 'ir.attachment'
 
-    @api.model
-    def _decode_edi_xml_p7m(self, filename, content):
-        """Decodes an xml.p7m into a list of one dictionary representing an attachment.
-        :returns:           A list with a dictionary.
+    def _decode_edi_l10n_it_edi(self, name, content):
+        """ Decodes a  into a list of one dictionary representing an attachment.
+            :returns:           A list with a dictionary.
         """
-        decoded_content = remove_signature(content)
-        if not decoded_content:
-            return
+        if not (decoded_content := remove_signature(content)):
```

Comment:
Walrus and not is like a Walnot (I don't know, it feels uncomfortable -> should not you use it inside the if ideally?)

Reply:
No... if there's no content, it's `return`. But in the code below, I want the `decoded_content`, not a `False` boolean.